precompiles/erc20/approve.go

```diff
@@ -168,13 +162,10 @@ func (p Precompile) DecreaseAllowance(
 	granter := contract.CallerAddress
 
 	// TODO: owner should be the owner of the contract
-	authorization, _, err := auth.CheckAuthzExists(ctx, p.AuthzKeeper, grantee, granter, SendMsgURL)
-	if err != nil {
-		return nil, err
-	}
 
-	// get allowance and ignore the error as it will be checked in the switch statement below
-	allowance, _ := GetAllowance(p.AuthzKeeper, ctx, grantee, granter, p.tokenPair.Denom)
+	// get authorization & allowance and ignore the error as
+	// it will be checked in the switch statement below.
+	authorization, allowance, err := GetAuthzAndAllowance(p.AuthzKeeper, ctx, grantee, granter, p.tokenPair.Denom)
```

Comment:
refactored this into the `GetAllowance` function because we were calling `CheckAuthzExists` twice here